FrontEnd/src/components/Browse.tsx

```diff
@@ -1,10 +1,193 @@
+        resolve(data);
+        //reject('Failed to load');
+      }, 2000);
+    });
+  }
+
+  const [error, setError] = useState(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [rows, setRows] = useState<any[]>([]);
+
+  useEffect(() => {
+    getData().then(
+      (data: any) => {
+        setIsLoaded(true);
+        setRows(data);
+      },
+      (error) => {
+        setIsLoaded(true);
+        setError(error);
+      }
+    );
+  }, []);
+
+  if (error) {
+    return <div>Error</div>;
+  } else if (!isLoaded) {
+    return (
+      <Grid
+        container
+        direction="column"
+        justifyContent="center"
+        alignItems="center"
+        sx={{ marginTop: 20 }}
+      >
+        <Box sx={{ fontWeight: 'bold', fontSize: '35px', mb: '30px' }}>
+          Loading Files...
+        </Box>
+        <CircularProgress size={60}></CircularProgress>
+      </Grid>
+    );
+  } else {
+    return (
+      <Grid
+        container
+        direction="row"
+        justifyContent="center"
+        alignItems="center"
+        sx={{ marginTop: 10 }}
+      >
+        <TableContainer
+          sx={{
+            maxHeight: '300px',
+            maxWidth: '75%',
+            border: 'solid',
+            borderColor: 'primary.main',
+            borderRadius: 2,
+          }}
+        >
+          <Table aria-label="simple table" stickyHeader>
+            <TableHead>
+              <TableRow>
+                <TableCell sx={{ fontWeight: 'bold' }}>Form</TableCell>
+                <TableCell sx={{ fontWeight: 'bold' }}>Quadrant</TableCell>
+                <TableCell sx={{ fontWeight: 'bold' }}>Date Entered</TableCell>
+                <TableCell sx={{ fontWeight: 'bold' }}>Validation</TableCell>
+                <TableCell align="center" sx={{ fontWeight: 'bold' }}>
+                  Actions
+                </TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {rows.map((row) => (
+                <TableRow key={row.file}>
+                  <TableCell>{row.file}</TableCell>
+                  <TableCell>{row.quadrant}</TableCell>
+                  <TableCell>{row.date}</TableCell>
+                  <TableCell>{row.validation}</TableCell>
+                  <TableCell align="center">
+                    <Button>
+                      <DownloadIcon></DownloadIcon>
+                    </Button>
+                    <Button>
+                      <EditIcon></EditIcon>
+                    </Button>
+                    <Button onClick={() => handleRemove(row.file)}>
+                      <DeleteIcon></DeleteIcon>
+                    </Button>
+                  </TableCell>
+                </TableRow>
+              ))}
+            </TableBody>
+          </Table>
+        </TableContainer>
+      </Grid>
+    );
+  }
+  /*
+  //TEMPORARY CODE BELOW
+  //Everything below is TEMPORARY data to test the UI for the table on browse page.
```

Comment:
Do we still need this temporary code? @Jaro05

Reply:
No we do not, I do not know why I didn't catch this!